packages/node_modules/@ciscospark/plugin-messages/src/messages.js

```diff
@@ -57,65 +55,105 @@ const Messages = SparkPlugin.extend({
    * @returns {Promise}
    */
   listen() {
-    return this.spark.internal.mercury.connect()
-      .then(() => this.listenTo(
-        this.spark.internal.mercury,
-        INCOMING_MESSAGES_CREATED,
-        (event) => this.onConversationActivityEvent(event)
-      ));
+    // Create a common envelope that we will wrap all events in
+    return createEventEnvelope(this.spark,
+      SDK_EVENT.TO_APP.RESOURCE.MESSAGES)
+      .then((envelope) => {
+        this.eventEnvelope = envelope;
```

Comment:
Do we have to worry about the timestamp never changing on this? I remember you explaining how you wanted it to be the same across certain cases, but I forget the nuance.

Reply:
On the webhook envelope, the timestamp is the the time that the user created (or modified) the webhook.   To me this is the equivilent as it represents the time that the user called listen().   This would be updated only if the user called listen() again.

Make sense?